library/Icingadb/Widget/HostListItemDetailed.php

```diff
@@ -4,11 +4,75 @@
 
 namespace Icinga\Module\Icingadb\Widget;
 
+use Icinga\Application\Icinga;
+use Icinga\Module\Icingadb\Common\ListItemDetailedLayout;
 use Icinga\Module\Icingadb\Compat\CompatPluginOutput;
+use Icinga\Module\Icingadb\Util\Perfdata;
+use Icinga\Module\Icingadb\Util\PerfdataSet;
 use ipl\Html\BaseHtmlElement;
+use ipl\Html\Html;
+use ipl\Html\HtmlString;
+use ipl\Web\Widget\Icon;
+use ipl\Web\Widget\StateBall;
 
-class HostListItemDetailed extends HostListItem
+class HostListItemDetailed extends BaseHostListItem
 {
+    use ListItemDetailedLayout;
+
+    protected $pie;
+
+    protected function getStateBallSize()
+    {
+        return StateBall::SIZE_LARGE;
+    }
+
+    protected function assembleFooter(BaseHtmlElement $footer)
+    {
+        $this->pie = HtmlString::create('');
+
+        $statusIcons = Html::tag('div', ['class' => 'status-icons']);
+
+        // ToDo(fs): Get `has_comments` from database
+        if ($this->item->comment->limit(1)->execute()->hasResult()) {
+            $statusIcons->add(new Icon('comments', ['title' => t('This item has been commented')]));
+        }
+
+        if (!$this->item->notifications_enabled) {
```

Comment:
Please add a space between `!` and what follows. There's another occurrence below.